chat/fe/src/views/RoomView.ts

```diff
@@ -46,17 +58,57 @@ export function RoomView(props: RoomViewProps) {
     const newMessage = Message(message);
     messageList.prepend(newMessage);
   });
+  onDeleteMessage(props.roomId, async (messageId) => {
+    const messageToDelete = byId(messageId);
+    messageToDelete.remove();
+    const isLastMessageInList =
+      messageList.firstElementChild.classList.contains('date-divider');
+
+    if (isLastMessageInList) {
+      messageList.removeChild(messageList.firstChild);
+      const messagesList = await getRoomMessages(props.roomId);
+      messages = messagesList.messages;
+    }
+  });
 
   function Message(props: MessageType) {
+    let dropdownOpen = false;
+
     const el = Div({
       class: 'message',
+      id: props._id,
     });
 
     setStyle(el, {
       display: 'flex',
       margin: '4px 0px',
     });
 
+    onMouseOver(el, (e) => {
+      e.stopPropagation();
+      if (messageButtonActive) {
+        return;
+      }
+
+      el.style.backgroundColor = '#f2f2f2';
+      const optionsButton = bySelector(el.lastElementChild, '.options-button');
```

Comment:
I'd probably like to avoid CSS selectors like this.  Ideally each component has an actual reference to each of its child elements.  I'll have to play with what you have to see how I'd prefer this to look

Reply:
Not sure if that's the place for this but added this in General so we can follow up on this https://github.com/adamjberg/engram/discussions/177